### PR TITLE
Bugfix/fix pegs

### PR DESCRIPTION
**Pull request by Lauraducky**
Fixes issues with iOS and web implementations of Webex

### CODE REVIEW

File: src/lib/parsers/xmlMessageParser.ts [hunk 1]
```diff
@@ -41,11 +50,11 @@ function getMessageXml(message : MessageObject) : xml.Document {
 	// Ensure it is encoded correctly
 	unescape.chars['&amp;'] = '&amp;';
 	let unencoded = unescape(message.html);
-	// XML parsing requires root node. If it doesn't exist, add one.
```
[the remark this answers is not in the document]
Comment (Lauraducky) [reply src/lib/parsers/xmlMessageParser.ts]: done

File: src/lib/response-triggers/default.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 import { MessageObject } from 'ciscospark/env';
+import xmlMessageParser from '../parsers/xmlMessageParser';
 
 export default class Default extends Trigger {
 	isToTriggerOn(message : MessageObject) : boolean {
-		let pattern = new RegExp('^' + constants.optionalMarkdownOpening + constants.mentionMe + 'ui');
-		return pattern.test(message.html);
+		let parsedMessage = xmlMessageParser.parseXmlMessage(message);
+		return parsedMessage.length > 0 && parsedMessage[0].name() === 'spark-mention'
+			&& xmlMessageParser.getPersonId(parsedMessage[0].attr('data-object-id').value()) === constants.botId;
```
[the remark this answers is not in the document]
Comment (Lauraducky) [reply src/lib/response-triggers/default.ts]: done

File: src/lib/parsers/xmlMessageParser.ts [hunk 3]
```diff
@@ -41,20 +67,23 @@ function getMessageXml(message : MessageObject) : xml.Document {
 	// Ensure it is encoded correctly
 	unescape.chars['&amp;'] = '&amp;';
 	let unencoded = unescape(message.html);
+
 	// XML parsing requires root node. If it doesn't exist, add one.
-	if (!unencoded.toLowerCase().startsWith('<p>')) {
+	if (!unencoded.toLowerCase().trim().startsWith('<p>') && !unencoded.toLowerCase().trim().startsWith('<div>')) {
```
Comment (jimcullenaus) [inline src/lib/parsers/xmlMessageParser.ts]: I hate this so much...

File: src/lib/parsers/xmlMessageParser.ts [hunk 4]
```diff
@@ -29,8 +29,34 @@ function parsePegMessage(message : MessageObject) : ParsedMessage {
...
 }
 
+function parseNonPegMessage(message : MessageObject) : ParsedMessage {
+	let children : xml.Element[] = parseXmlMessage(message);
+	let parsedMessage : ParsedMessage = {
+		fromPerson: message.personId,
+		botId: children.length > 0 && children[0].name() === 'spark-mention' ? getPersonId(children[0].attr('data-object-id').value()) : null,
+		children,
+		command: children.reduce((a, child, index) => {
+			// first three children should be mentions or command words
```
Comment (Evangelinexx) [inline src/lib/parsers/xmlMessageParser.ts]: comment is copy-pasted innacurately

File: src/lib/parsers/xmlMessageParser.ts (hunk 3 above)
Comment (Evangelinexx) [reply src/lib/parsers/xmlMessageParser.ts]: Yeah... this shouldn't need to exist but Cisco is dumb.

File: src/lib/parsers/xmlMessageParser.ts (hunk 4 above)
Comment (Lauraducky) [reply src/lib/parsers/xmlMessageParser.ts]: fixed. goddamnit

File: src/constants.ts [hunk 5]
```diff
@@ -13,5 +13,6 @@ export default {
 	mentionMe: `<spark-mention data-object-type="person" data-object-id="${process.env.BOT_ID}">${process.env.BOT_NAME}<\/spark-mention>`,
 	optionalMarkdownOpening: '(?:<p>)?',
 	optionalMarkdownEnding: '(?:<\/p>)?',
-	optionalSpace: '(?: )*'
+	optionalSpace: '(?: )*',
+	sparkTokenPrefix: 'ciscospark://us/PEOPLE/'
```
Comment (jimcullenaus) [inline src/constants.ts]: ffs cisco

**Pull request merged**